### PR TITLE
photon UDP support

### PR DESCRIPTION
**Pull request by m-mcgowan**
### Problem

UDP Cloud Communications on the Photon/P1. This was not working correctly after refactoring mbedtls in the system firmware and the Gen2/Gen3 integration.

### Solution

- Allow the standard mbedtls config to be used (not the special photon one) when UDP is enabled.
- Weaken the duplicated mbedtls functions from the BCEL library in WICED so that we don't get duplicate 

### Steps to Test
- Edit `hal_platform_compat.h` to enable `HAL_PLATFORM_CLOUD_UDP` on the Photon/P1. 
- Flash monolithic firmware (It does not build with separate system parts.)
- Power on the photon, verify that it connects to the cloud. 
- Verify the handshake is visible via fast blinking cyan LED
- Hit reset
- Verify the handshake is no longer visible and that the device almost immediately connects to the cloud. This is a symptom of UDP session resumption. 
- NB: I did not test that TCP communications still works. UDP is enabled by default with this build. (Verify!)
- Verify WPA Enterprise authentication still works after this change.

### Example App

```c
void setup() {
  /* A minimal example app is super helpful 
   * for testing new features and fixes. 
   * A link to a Docs PR is even better!
   */
}

void loop() {

}
```

### References

Links to the Community, Docs, Other Issues, etc..

---

### Completeness

- [x] User is totes amazing for contributing!
- [ ] Contributor has signed CLA ([Info here](https://github.com/spark/firmware/blob/develop/CONTRIBUTING.md))
- [ ] Problem and Solution clearly stated
- [ ] Run unit/integration/application tests on device
- [ ] Added documentation
- [ ] Added to CHANGELOG.md after merging (add links to docs and issues)

### CODE REVIEW

File: crypto/inc/mbedtls_config_default.h [hunk 1]
```diff
@@ -21,10 +21,19 @@
  *  This file is part of mbed TLS (https://tls.mbed.org)
  */
 
-#if PLATFORM_ID == 6 || PLATFORM_ID == 8
+#include "hal_platform.h"
+
+#if PLATFORM_ID==6 || PLATFORM_ID==8
+#define PLATFORM_WICED 1
```
Comment (avtolstoy) [inline crypto/inc/mbedtls_config_default.h]: `HAL_PLATFORM_WICED` in `hal_platform.h` perhaps?

We should also get rid of these `mbedlts_config_default.h`/`mbedtls_config_photon.h` in the future and use the same scheme for Gen 2 devices as for Gen 3: platform-specific `mbedtls_config_platform.h` in platform-specific HAL folder.

File: hal/src/photon/lib/BESL.ARM_CM3.release.a.weaken [hunk 2]
```diff
@@ -37,3 +37,24 @@ rsa_pkcs1_verify
...
 rsa_check_privkey
+mbedtls_ccm_init
+mbedtls_ccm_setkey
+mbedtls_ccm_free
+mbedtls_ccm_encrypt_and_tag
+mbedtls_ccm_auth_decrypt
+mbedtls_cipher_info_from_type
+mbedtls_cipher_info_from_string
+mbedtls_cipher_info_from_values
+mbedtls_cipher_init
```
Comment (avtolstoy) [inline hal/src/photon/lib/BESL.ARM_CM3.release.a.weaken]: We should definitely verify that WPA Enterprise authentication still works after this change.

File: crypto/inc/mbedtls_config_default.h [hunk 3]
```diff
@@ -21,10 +21,19 @@
  *  This file is part of mbed TLS (https://tls.mbed.org)
  */
 
-#if PLATFORM_ID == 6 || PLATFORM_ID == 8
+#include "hal_platform.h"
+
+#if PLATFORM_ID==6 || PLATFORM_ID==8
```
Comment (avtolstoy) [inline crypto/inc/mbedtls_config_default.h]: Nitpicking: spaces around `=`

File: hal/src/photon/lib/BESL.ARM_CM3.release.a.weaken (hunk 2 above)
Comment (technobly) [reply hal/src/photon/lib/BESL.ARM_CM3.release.a.weaken]: Added to Steps to test

**Comment by m-mcgowan** [comment]
Closing out. Not sure if it is still relevant.

**Pull request closed without merging**